File: android/src/main/java/io/mosip/tuvali/cryptography/CryptoBox.java
package io.mosip.tuvali.cryptography;

interface CryptoBox {
    int INITIALISATION_VECTOR_LENGTH = 12;
    byte[] getPublicKey();
    CipherPackage createCipherPackage(byte[] otherPublicKey, String selfInfo, String receipientInfo, byte[] ivBytes);
}

File: ios/crypto/CryptoBox/CryptoBox.swift
//
//  CryptoBox.swift
//  Openid4vpBle
//
//  Created by Shaik mohammed Jaffer on 22/12/22.
//  Copyright © 2022 Facebook. All rights reserved.
//

import Foundation
import CryptoKit

protocol CryptoBox {
    func createCipherPackage(otherPublicKey: Data, senderInfo: String, recieverInfo: String, ivBytes: Data) -> CipherPackage
    func getPublicKey() -> Data
}

